src/rpc/mempool.cpp

```diff
@@ -377,6 +427,63 @@ UniValue MempoolToJSON(const CTxMemPool& pool, bool verbose, bool include_mempoo
     }
 }
 
+static RPCHelpMan getmempoolfeesize()
+{
+    return RPCHelpMan{"getmempoolfeesize",
+        "Returns fee/size data for the whole mempool.",
+        {},
+        {},
+        RPCExamples{
+            HelpExampleCli("getmempoolfeesize", "")
+            + HelpExampleRpc("getmempoolfeesize", "")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+        {
+            const CTxMemPool& mempool = EnsureAnyMemPool(request.context);
+            LOCK(mempool.cs);
+
+            UniValue result(UniValue::VARR);
+
+            // TODO: refactor so that we're not just copying this from the miner.
+            std::vector<Cluster::HeapEntry> heap_chunks;
+            // Initialize the heap with the best entry from each cluster
+            for (const auto & [id, cluster] : mempool.m_cluster_map) {
+                if (!cluster->m_chunks.empty()) {
+                    heap_chunks.emplace_back(cluster->m_chunks.begin(), cluster.get());
+                }
+            }
+            // Define comparison operator on our heap entries (using feerate of chunks).
+            auto cmp = [](const Cluster::HeapEntry& a, const Cluster::HeapEntry& b) {
+                return a.first->fee*b.first->size < b.first->fee*a.first->size;
```

Comment:
0b284b5fd29c06d46c1ec60ea7e1bcd07f36feb1 : Could this use `CompareMiningScore` or does it have a different goal?